cmsBuild

```diff
@@ -3782,6 +3805,42 @@ def rpmCacheUpdate(opts, cache=None, rpm_env=None):
             cache[n] = [r, c]
     return
 
+def checkOptionsLocalSource(opts):
+    """
+    checks if opts.localSources is correct format `package:source=directory` and process it
+    Could be made callback function in advancedBuildOptions.add_option
+    """
+    # check this option is given
+    if opts.localSources is None:
+        return  # opts.localSources is empty, return
+    sourceOptionDict = {}
+    for sourceOption in opts.localSources:
+        # check if sourceOption in correct format
+        if '=' not in sourceOption :
+            fatal("ERROR: --source option is incorrectly given. Should be 'package:source=directory'.")
+        pkg_and_source, dir_path = sourceOption.split("=")
+        pkg = source = None
+        if ":" in pkg_and_source:
+            pkg, source = pkg_and_source.split(":")
+        else:
+            pkg = pkg_and_source
+            source = "Source"
+        #check if given directory path is correct
+        if not exists(dir_path):
+            fatal("WARNING: the following direcotry for package %s and source %s does not exist: %s" % (pkg, source, dir_path))
+        if not os.path.isdir(dir_path):
```

Comment:
as discussed, we should allow to pass a file path too. so check about directory and empty directory can be relaxed here

Reply:
https://github.com/gudrutis/pkgtools/commit/0ec3662095d9de63c93c99e9e36d187f2223c3c8

In this commit I allow files and directories, but if it is an empty directory, I will fail it.